src/webgpu/api/validation/encoding/cmds/render/draw.spec.ts

```diff
@@ -142,7 +178,96 @@ In this test we test that a small buffer bound to unused buffer slot won't cause
+  .params(u =>
+    u //
+      .combine('smallIndexBuffer', [false, true])
+      .combine('smallVertexBuffer', [false, true])
+      .combine('smallInstanceBuffer', [false, true])
+      .beginSubcases()
+      .combine('drawType', ['draw', 'drawIndexed', 'drawIndirect', 'drawIndexedIndirect'] as const)
+      .unless(
+        // Always provide index buffer of enough size if it is used by indexed draw
+        p =>
+          p.smallIndexBuffer &&
+          (p.drawType === 'drawIndexed' || p.drawType === 'drawIndexedIndirect')
+      )
+      .combine('bufferOffset', [0, 4])
+      .combine('boundSize', [0, 1])
+  )
+  .fn(async t => {
+    const {
+      smallIndexBuffer,
+      smallVertexBuffer,
+      smallInstanceBuffer,
+      drawType,
+      bufferOffset,
+      boundSize,
+    } = t.params;
+    const renderPipeline = t.createNoOpRenderPipeline();
+    const bufferSize = bufferOffset + boundSize;
+    const smallBuffer = t.createBufferWithState('valid', {
+      size: bufferSize,
+      usage: GPUBufferUsage.INDEX | GPUBufferUsage.VERTEX,
+    });
+    // An index buffer of enough size, used if smallIndexBuffer === false
+    const indexBuffer = t.createBufferWithState('valid', {
+      size: 200,
+      usage: GPUBufferUsage.INDEX,
+    });
+
+    for (const useBundle of [false, true]) {
+      for (const setBufferBeforePipeline of [false, true]) {
+        const commandBufferMaker = t.createEncoder(useBundle ? 'render bundle' : 'render pass');
+        const renderEncoder = commandBufferMaker.encoder;
+
+        if (!setBufferBeforePipeline) {
+          renderEncoder.setPipeline(renderPipeline);
+        }
+
+        if (drawType === 'drawIndexed' || drawType === 'drawIndexedIndirect') {
+          // Always use large enough index buffer for indexed draw. Index buffer OOB validation is
+          // tested in index_buffer_OOB.
+          renderEncoder.setIndexBuffer(indexBuffer, 'uint16', 0, 200);
+        } else if (smallIndexBuffer) {
+          renderEncoder.setIndexBuffer(smallBuffer, 'uint16', bufferOffset, boundSize);
+        }
+        if (smallVertexBuffer) {
+          renderEncoder.setVertexBuffer(1, smallBuffer, bufferOffset, boundSize);
+        }
+        if (smallInstanceBuffer) {
+          renderEncoder.setVertexBuffer(7, smallBuffer, bufferOffset, boundSize);
+        }
+
+        if (setBufferBeforePipeline) {
+          renderEncoder.setPipeline(renderPipeline);
+        }
+
+        if (drawType === 'draw' || drawType === 'drawIndirect') {
+          const drawParam: DrawParameter = {
+            vertexCount: 100,
+            instanceCount: 100,
+            firstVertex: 100,
+            firstInstance: 100,
+          };
+
+          callDraw(t, renderEncoder, drawType, drawParam);
+        } else {
+          const drawParam: DrawIndexedParameter = {
+            indexCount: 100,
+            instanceCount: 100,
+            firstIndex: 0,
+            baseVertex: 100,
+            firstInstance: 100,
```

Comment:
I see you are using '100' almost everywhere. How about defining a constant instead of always using a magic number?

Reply:
Done.